framework/src/meshgenerators/LowerDBlockFromSidesetGenerator.C

```diff
@@ -30,19 +30,27 @@ validParams<LowerDBlockFromSidesetGenerator>()
   params.addParam<SubdomainID>("new_block_id", "The lower dimensional block id to create");
   params.addParam<SubdomainName>("new_block_name",
                                  "The lower dimensional block name to create (optional)");
-  params.addRequiredParam<std::vector<boundary_id_type>>(
-      "sidesets", "The sidesets from which to create the new block");
+  params.addParam<std::vector<boundary_id_type>>(
+      "sidesets",
+      "The sidesets from which to create the new block. Either sideset_names or sidesets must be "
+      "specified");
+  params.addParam<std::vector<BoundaryName>>(
+      "sideset_names",
+      "The sideset names from which to create the new block. Either sideset_names or sidesets must "
+      "be specified");
```

Comment:
BoundaryNames are supposed to handle both strings and IDs simultaneously so that users would never have to worry about differentiating between them. We have tests where you can do things like "bottom top 3" and the system will work just fine. This works because the methods on MooseMesh are designed to take that "mixed" vector and convert them to a list of IDs for use internally. If this *doesn't* work in the MeshGenerator system, then that should be fixed. Have you tried using those methods on MooseMesh? If they don't work, then it's worth fixing so we don't introduce this inconsistent syntax.

Reply:
hm, running
```
~/projects/moose/test/tests/meshgenerators/lower_d_block_generator$ ../../../moose_test-opt -i lower_d_block_generator.i Mesh/lower_d_block/sidesets='0 bottom 1 2 3'
```

Results in 
```
*** ERROR ***
/Users/topher/projects/moose/test/tests/meshgenerators/lower_d_block_generator/lower_d_block_generator.i:13: cannot convert 'bottom' to int
```

Reply:
I was following similar treatment in `RenameBoundaryGenerator` which as a vector of `boundary_id_type` and `BoundaryName`, and coverts the latter to an id during `RenameBoundaryGenerator::generate()`

Reply:
Yeah - I noticed `RenameBoundaryGenerator` was doing it wrong too :)

These are the converter functions here:
https://github.com/idaholab/moose/blob/next/framework/include/mesh/MooseMesh.h#L574-L605

I'm not 100% positive that they work with the Generators though. However, if they don't, the need to and I'm happy to look at making that work. The interface is to pass in your mixed entries and get back all IDs. See if that works really quick and let me know.

Reply:
Yup! That did it!